src/VisualStudio/Core/Def/StackTraceExplorer/StackFrameViewModel.cs

```diff
@@ -0,0 +1,310 @@
+                var textBetweenLength = frameWithFile.FileSpan.Start - argEndIndex;
+                var textBetweenSpan = new TextSpan(argEndIndex, textBetweenLength);
+                if (textBetweenSpan.Length > 0)
+                {
+                    var textBetween = _frame.OriginalText.Substring(textBetweenSpan.Start, textBetweenSpan.Length);
+                    yield return new Run(textBetween);
+                }
+
+                var fileText = _frame.OriginalText[frameWithFile.FileSpan.Start..frameWithFile.FileSpan.End];
+                var fileHyperlink = new Hyperlink();
+                fileHyperlink.Inlines.Add(MakeClassifiedRun(ClassificationTypeNames.Text, fileText));
+                fileHyperlink.RequestNavigate += (s, e) => NavigateToFile();
+                fileHyperlink.Click += (s, e) => NavigateToFile();
+                yield return fileHyperlink;
+
+                var end = frameWithFile.FileSpan.End;
+                if (end < _frame.OriginalText.Length)
+                {
+                    yield return MakeClassifiedRun(ClassificationTypeNames.Text, _frame.OriginalText[..end]);
+                }
+            }
+            else
+            {
+                if (argEndIndex < _frame.OriginalText.Length)
+                {
+                    yield return MakeClassifiedRun(ClassificationTypeNames.Text, _frame.OriginalText[argEndIndex..]);
+                }
+            }
+        }
+
+        private Document? GetDocument(out int lineNumber)
+        {
+            if (_cachedDocument is not null)
+            {
+                lineNumber = _cachedLineNumber;
+                return _cachedDocument;
+            }
+
+            var potentialMatches = GetFileMatches(out _cachedLineNumber);
+            if (potentialMatches.Any())
+            {
+                _cachedDocument = potentialMatches.First();
+            }
+
+            lineNumber = _cachedLineNumber;
+            return _cachedDocument;
+        }
+
+        private async Task<ISymbol?> GetSymbolAsync(CancellationToken cancellationToken)
+        {
+            if (_cachedSymbol is not null)
+            {
+                return _cachedSymbol;
+            }
+
+            _cachedSymbol = await _frame.ResolveSymbolAsync(_workspace.CurrentSolution, cancellationToken).ConfigureAwait(false);
+            return _cachedSymbol;
+        }
+
+        private ImmutableArray<Document> GetFileMatches(out int lineNumber)
+        {
+            var fileLineResult = _frame as ParsedFrameWithFile;
+            Contract.ThrowIfNull(fileLineResult);
+
+            var fileText = _frame.OriginalText.Substring(fileLineResult.FileSpan.Start, fileLineResult.FileSpan.Length);
+            Debug.Assert(fileText.Contains(':'));
+
+            var splitIndex = fileText.LastIndexOf(':');
+
+            var fileName = fileText[..splitIndex];
+            var lineNumberText = fileText[(splitIndex + 1)..];
+
+            var numberRegex = new Regex("[0-9]+");
```

Comment:
Isn't `int.TryParse` sufficient at this point? alternatively, use a regex but move the `:` parsing into it too.